glean-core/tests/string_list.rs

```diff
@@ -23,17 +26,142 @@ fn list_can_store_multiple_items() {
     assert!(!snapshot.contains(r#""first""#));
     assert!(!snapshot.contains(r#""second""#));
     assert!(snapshot.contains(r#""third""#));
+    assert!(!snapshot.contains(r#""fourth""#));
+
+    list.add(&glean, "fourth");
+    let snapshot = glean.snapshot("core", false);
+    assert!(snapshot.contains(r#""local.list": ["#));
+    assert!(!snapshot.contains(r#""first""#));
+    assert!(!snapshot.contains(r#""second""#));
+    assert!(snapshot.contains(r#""third""#));
+    assert!(snapshot.contains(r#""fourth""#));
+}
+
+#[test]
+fn stringlist_serializer_should_correctly_serialize_stringlists() {
+    let (_t, tmpname) = tempdir();
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID, true).unwrap();
+        let metric = StringListMetric::new(CommonMetricData {
+            name: "string_list_metric".into(),
+            category: "telemetry.test".into(),
+            send_in_pings: vec!["store1".into()],
+            disabled: false,
+            lifetime: Lifetime::User,
+        });
+        metric.set(&glean, vec!["test_string_1".into(), "test_string_2".into()]);
+    }
+
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID, true).unwrap();
+
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), "store1", true)
+            .unwrap();
+        assert_eq!(
+            json!({"string_list": {"telemetry.test.string_list_metric": ["test_string_1", "test_string_2"]}}),
+            snapshot
+        );
+    }
+}
+
+#[test]
+fn set_properly_sets_the_value_in_all_stores() {
+    let (glean, _t) = new_glean();
+    let store_names: Vec<String> = vec!["store1".into(), "store2".into()];
+
+    let metric = StringListMetric::new(CommonMetricData {
+        name: "string_list_metric".into(),
+        category: "telemetry.test".into(),
+        send_in_pings: store_names.clone(),
+        disabled: false,
+        lifetime: Lifetime::Ping,
+    });
+
+    metric.set(&glean, vec!["test_string_1".into(), "test_string_2".into()]);
+
+    for store_name in store_names {
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), &store_name, true)
+            .unwrap();
+
+        assert_eq!(
+            json!({"string_list": {"telemetry.test.string_list_metric": ["test_string_1", "test_string_2"]}}),
+            snapshot
+        );
+    }
+}
+
+#[test]
+fn long_string_values_are_truncated() {
+    let (glean, _t) = new_glean();
+
+    let metric = StringListMetric::new(CommonMetricData {
+        name: "string_list_metric".into(),
+        category: "telemetry.test".into(),
+        send_in_pings: vec!["store1".into()],
+        disabled: false,
+        lifetime: Lifetime::Ping,
+    });
+
+    let test_string = "0123456789".repeat(20);
+    metric.add(&glean, test_string.clone());
+
+    // Ensure the string was truncated to the proper length.
+    assert_eq!(
+        format!("[\"{}\"]", test_string[..50].to_string()),
```

Comment:
Instead of constructing the JSON string here ourselves how about we can expose a proper `test_get_value` function that returns `Vec<String>`